lib/attrsets.nix

```diff
@@ -56,9 +56,7 @@ rec {
   */
   setAttrByPath = attrPath: value:
     if attrPath == [] then value
```

Comment:
This conditional seems to have become redundant. I don't expect it to be particularly frequent, so it wouldn't surprise me if removing it improves performance. That might not show in the stats though. Not sure if this is relevant after the other suggestion, so try that one first.

Reply:
yeah, it is redundant as the conditional in the definition of `foldr` is equivalent. thanks.